### ISSUE TITLE
Add PyPy support to CircleCI

### ISSUE THREAD

**Issue by dhermes** [dhermes/bezier#4]
Remnants of failed attempt:
https://github.com/dhermes/bezier/blob/a6b30013cccf8998e5b62e90a74db9f039e1b7a9/tox.ini
https://github.com/dhermes/bezier/blob/a6b30013cccf8998e5b62e90a74db9f039e1b7a9/circle.yml

Essentially, the NumPy / SciPy / Matplotlib story isn't really developed until later version of PyPy (4.1 and >= 5.0):
- http://pypy.org/download.html#installing-numpy
- https://morepypy.blogspot.ch/2016/06/pypy2-v53-released-major-c-extension.html

----

It's worth noting, that on my local install:

```
$ pypy --version
Python 2.7.10 (5.3.1+dfsg-1~ppa1~ubuntu14.04, Jun 19 2016, 15:09:54)
[PyPy 5.3.1 with GCC 4.8.4]
```

running the tests with PyPy are comically slow:

```
$ tox -e functional
...
======= 57 passed in 1.12 seconds =======
...
$ tox -e functional-pypy
...
======= 57 passed in 6.19 seconds =======
...
$ tox -e py27
...
======= 204 passed in 3.72 seconds =======
...
$ tox -e pypy
...
======= 203 passed, 1 skipped in 28.25 seconds =======
...
```

----

It's common to have a `requirements.txt` file that uses environment markers as follows:

```
numpy; platform_python_implementation != "PyPy"
git+https://bitbucket.org/pypy/numpy.git; platform_python_implementation == "PyPy"
```

**Comment by dhermes** [comment]
https://bitbucket.org/pypy/numpy.git seems to be out of date, **however** `pypy -m pip install numpy` "just works" now, so that is not so bad (might be possible to build a wheel to speed up the process?)

Also, now that the tests run on CircleCI 2.0, it's in my control to just build PyPy into the docker image. Probably helpful

- https://hub.docker.com/library/pypy/
- https://github.com/docker-library/pypy/blob/ae4d6e7d038eb080a55fc3d8d91594505e0f4030/2/Dockerfile

**Comment by dhermes** [comment]
Fixed by 110992a77fdccf170f859b90b0f85f057edc41dd and 1bf34a2ed89bcf442e99caa16e56cdcfa6149265
